### PR TITLE
Backport fix for LOGBACK-1027 from 1.3.x.

### PR DESCRIPTION
**Pull request by ellebrecht**
For users of Logback 1.2.x suffering from LOGBACK-1027, provide a solution by backporting the fix from the master branch.

### CODE REVIEW

File: logback-classic/src/main/java/ch/qos/logback/classic/spi/ThrowableProxy.java [hunk 1]
```diff
@@ -29,59 +32,52 @@
     int commonFrames;
     private ThrowableProxy cause;
     private ThrowableProxy[] suppressed = NO_SUPPRESSED;
+    private final Set<Throwable> alreadyProcessedSet;
```
Comment (klya) [inline logback-classic/src/main/java/ch/qos/logback/classic/spi/ThrowableProxy.java:35]: Actually this field looks useless.

**Comment by klya** [comment]
@ceki could you merge this? logback 1.2.6 is currently in production and LOGBACK-1027 hurts us a lot.

**Comment by mrmegapolys** [comment]
Also asking for merge, affects our production as well.

**Comment by ceki** [comment]
You are welcome to sponsor any bug fix or feature. Please see https://github.com/sponsors/qos-ch/sponsorships?tier_id=77436&preview=false

**Comment by zeldigas** [comment]
Nice price for backport, but I got your point ;)

**Comment by harsh-tamr** [comment]
Can this be merged?

**Comment by ceki** [comment]
There are so many issues to attend to. This issue will be dealt with eventually without any sponsoring. However, if you wish to sponsor a release as a gesture of goodwill, then do not hesitate to do so. All I can say is that it won't hurt.

**Comment by madhead** [comment]
Is it possible to partially sponsor this? I won't be able to convice my company to make a [one-time payment](https://github.com/sponsors/qos-ch/sponsorships?tier_id=77436&preview=false), but I could send a few bucks on my own. And I guess others here could contribute to the $800 goal too! Together we could afford it, don't we?

**Comment by ceki** [comment]
This PR removes the compatibility layer with Java 1.6 with no mention of the removal. This PR, were it applied as is, would break software running under Java 1.6.

**Comment by ceki** [comment]
Fixed in 1.2.11 released earlier today.

**Pull request closed without merging**